examples/09-averaging/02-elemental_nodal_to_nodal_max_contribution.py

```diff
@@ -0,0 +1,334 @@
+    value.
+
+    Parameters
+    ----------
+    meshed_region: MeshedRegion
+    number_of_components: int
+        Number of components the output field contains.
+        Default is 6 (e.g. stress field).
+
+    Returns
+    -------
+    elemental_nodal_field: Field
+
+    """
+    if number_of_components == 1:
+        nat = dpf.natures.scalar
+    elif number_of_components == 3:
+        nat = dpf.natures.vector
+    elif number_of_components == 6:
+        nat = dpf.natures.symmatrix
+    else:
+        raise Exception(f"{number_of_components} number of components not supported")
+    elem_nod_field = dpf.Field(nature=nat, location=dpf.locations.elemental_nodal)
+    field_def = elem_nod_field.field_definition
+    field_def.name = "stress"
+    elems_scop = meshed_region.elements.scoping
+    connectivity_field = meshed_region.elements.connectivities_field
+    for el_id in elems_scop.ids:
+        ele_ind = elems_scop.index(el_id)
+        el_connectivity = connectivity_field.get_entity_data(ele_ind)
+        field_val = []
+        for nod_ind in el_connectivity:
+            i = 0
+            while i < number_of_components:
+                field_val.append(20.0 * ele_ind)
+                i += 1
+        if len(field_val) > 0:
+            elem_nod_field.append(field_val, el_id)
+
+    return elem_nod_field
+
+
+###############################################################################
+# Define averaging method:
+def averaging_using_max_value(elemental_nodal_field, b_compute_max=True):
+    """Computes an averaging based on the highest value for a specific node.
+    For elemental nodal field, each node has several contributions (one per element).
+    The averaged field is nodal and for each node, the taken value is the maximum of
+    the contributions.
+    For example, node 1 is connected to elements 1,2,3 and 4. The maximum nodal
+    stress value at node 1 among all the element's contributions (maximum of
+    stress at node 1 among elements 1,2,3 and 4) is the contribution of element
+    3. So the elemental nodal value for element 3, node 1 is taken and placed in
+    the nodal averaged field. If b_compute_max is set to False, then minimum value
+    is taken.
+
+    This algorithm is made for solid or surface elements (shell are not supported).
+    Results are mid side nodes are not taken into account.
+    Only elemental nodal fields with one component (e.g. von mises
+    stress, sxx) are handled.
+
+    Parameters
+    ----------
+    elemental_nodal_field: Field
+        Elemental nodal field the averaging must be applied on.
+    b_compute_max: bool
+        Default is True. If set to True, considers the maximum value
+        for averaging. If set to False, considers the minimum value
+        for averaging.
+
+    Returns
+    -------
+    averaged_field: Field
+
+    """
+    # initial checks
+    ncomp = elemental_nodal_field.component_count
+    if ncomp != 1:
```

Comment:
@anslpa , why do we  restrict to one component? can't we have a max per component?

Reply:
@rlagha it would imply to have one output field per component, right ?

Reply:
Done